source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -231,6 +233,63 @@ def image(
     return images.clone()
 
 
+def image_features(
+    env: ManagerBasedEnv,
+    sensor_cfg: SceneEntityCfg = SceneEntityCfg("tiled_camera"),
+    data_type: str = "rgb",
+    convert_perspective_to_orthogonal: bool = True,
+    model_name: str = "Theia",
+    model_zoo_cfg: dict | None = None,
+) -> torch.Tensor:
+    """Extracted image features with a frozen encoder from Images of a specific datatype from the camera sensor.
+
+    Args:
+        env: The environment the cameras are placed within.
+        sensor_cfg: The desired sensor to read from. Defaults to SceneEntityCfg("tiled_camera").
+        data_type: The data type to pull from the desired camera. Defaults to "rgb".
+        model_name: The name of which model to use from the model_zoo_cfg to use to extract features.
+        model_zoo_cfg: A dictionary with string keys and callable values. Should include "model",
+            (mapped to a callable with no arguments to return the model), "preprocess" (mapped to
+            a callable which consumes the images and returns the preprocessed images),
+            and "inference" (mapped to a callable that provided the model, and the preproccessed images,
+            returns the features.)
+
+    Returns:
+        The features from the images produced at the last timestep
+    """
+    if not hasattr(image_features, "model_zoo"):
+        image_features.model_zoo = {}
```

Comment:
How are you binding this to a function? o.O

Reply:
Function attributes are pretty cool: 

https://docs.python.org/3/reference/datamodel.html

https://peps.python.org/pep-0232/

"Function objects also support getting and setting arbitrary attributes, which can be used, for example, to attach metadata to functions. Regular attribute dot-notation is used to get and set such attributes."